index.js

```diff
@@ -181,7 +193,7 @@ module.exports = function(input, map) {
         thunk: true,
         create: true,
       })
-      cachePath = thunk("data.json") || os.tmpdir() + "/data.json"
+      cachePath = thunk("data.json") || cacheFallback
```

Comment:
Both this line and the previous line (where you declare `thunk`) can throw.
You need to move them inside a `try`/`catch` and apply the same strategy of fallback.

Right now `|| cacheFallback` is not doing anything because if something's bad, the function will throw rather than return `null`.